### PR TITLE
Add dnsHome.de API

### PR DESCRIPTION
**Pull request by Bodenhaltung**
Add dnsHome.de API, DynDNS Subdomain Service.

### CODE REVIEW

**Comment by Bodenhaltung** [comment]
How can I disable the Github Action check for "www." this is not necessary/supported.

**Comment by Bodenhaltung** [comment]
Is there anything I can do here?

**Comment by Neilpang** [comment]
why is `www` not supported ?

**Comment by Bodenhaltung** [comment]
> 

It is a Dynamic DNS Service, the user “owns” only the Subdomain name, without the „www“ prefix.

**Comment by Neilpang** [comment]
please define a secret  `TEST_DNS_NO_SUBDOMAIN=1`, and rerun the tests.

**Comment by Bodenhaltung** [comment]
> please define a secret `TEST_DNS_NO_SUBDOMAIN=1`, and rerun the tests.

I defined the secret but it failed again. It looks like `TEST_DNS_NO_SUBDOMAIN` is not set in env. I also rebased and pushed to "force" a new Github Action run, but it doesn't work. What do I failed to see?

**Comment by Neilpang** [comment]
please rebase to the latest dev branch and run the tests again.

**Comment by Bodenhaltung** [comment]
> please rebase to the latest dev branch and run the tests again.

Ok, docker Action succeed, but MacOS failed, looks like `TEST_DNS_NO_SUBDOMAIN* doesn't exist there.

**Comment by Bodenhaltung** [comment]
> rebase and try again.

Thanks, now all tests passed exept "Solaris": `Error: The process '/usr/bin/ssh' failed with exit code ***`

**Comment by Neilpang** [comment]
please rerun it.  All the CI check MUST be passing before it can be merged.

**Comment by Neilpang** [comment]
you don't need to make empty push.  Just click the "rerun" button in your actions page.

**Comment by Bodenhaltung** [comment]
> you don't need to make empty push. Just click the "rerun" button in your actions page.

Sure, I jsut rebased.

But failed again for Slowlaris. :p

**Comment by Neilpang** [comment]
rerun it, until it's passing or you figure out why it fails.

**Comment by Bodenhaltung** [comment]
> rerun it, until it's passing or you figure out why it fails.

It looks like `curl` is fishy from the solaris image, but I can't see the exact (curl) error neither the "response", all reducted or written to local logs.

Are you sure there is no generic problem with solaris image?
If I look into the other PRs here, a few stucking also with the Solaris check.

**Comment by Neilpang** [comment]
That's why I asked you to rebase to the latest dev branch.
The solaris image was fixed, and it's passing on the dev branch:
https://github.com/acmesh-official/acme.sh/actions/runs/1572760483

**Comment by Bodenhaltung** [comment]
> That's why I asked you to rebase to the latest dev branch. The solaris image was fixed, and it's passing on the dev branch: https://github.com/acmesh-official/acme.sh/actions/runs/1572760483

rebase from dev was done, the version change commit is in the forked branch.

I see no error, exept:
```
[Thursday, December ***6, 202*** at  6:00:***0 PM UTC] _post_url='***www.dnshome.de/dyndns.php'
[Thursday, December ***6, 202*** at  6:00:*** PM UTC] _CURL='curl --silent --dump-header /root/.acme.sh/http.header  -L  -g '
[Thursday, December ***6, 202*** at  6:00:***2 PM UTC] **;3***mPlease refer to https://curl.haxx.se/libcurl/c/libcurl-errors.html for error code: ***
[Thursday, December ***6, 202*** at  6:00:***2 PM UTC] _ret='***'
[Thursday, December ***6, 202*** at  6:00:***3 PM UTC] _hcode='***'
[Thursday, December ***6, 202*** at  6:00:***4 PM UTC] **;3***merror acme=add&txt=uPXqG2bY-EEkwSirNpkMKol7iqx-mdLzaiC***eDWc_uY
[Thursday, December ***6, 202*** at  6:00:***4 PM UTC] **;3***mError
[Thursday, December ***6, 202*** at  6:00:***4 PM UTC] **;3***m
[Thursday, December ***6, 202*** at  6:00:***5 PM UTC] **;3***mError add txt for domain:_acme-challenge.***
[Thursday, December ***6, 202*** at  6:00:***5 PM UTC] _on_issue_err
[Thursday, December ***6, 202*** at  6:00:***5 PM UTC] **;3***mPlease check log file for more details: le_test_dnsapi.log
```

But not really helpfull because reducted. I didn't see the plain http header neither the http answer or "le_test_dnsapi.log". It makes debugging imposible, even the other checks are passing without errors, as I said before, looks like a `curl` problem in solaris image, but without the error code or $something helpfull about the error it impossible to debug.

**Comment by Neilpang** [comment]
you can use this project to get a solaris vm to debug:
https://github.com/vmactions/shell-solaris

**Comment by Neilpang** [comment]
can you figure ou how to upgrade the root certificates on solaris?

**Comment by Bodenhaltung** [comment]
Looks like fixed now, checks succeeded.

**Comment by Bodenhaltung** [comment]
Hmm, why skipped? Can you trigger a rerun? My runs are green

**Comment by Neilpang** [comment]
```

 export DNSHOME_Subdomain=""
 export DNSHOME_SubdomainPassword=""

```


Is one subdomain password for one subdomain only?

**Comment by Bodenhaltung** [comment]
> ```
> 
>  export DNSHOME_Subdomain=""
>  export DNSHOME_SubdomainPassword=""
> ```
> 
> 
>      
>   
> 
> Is one subdomain password for one subdomain only?

Yes.

File: dnsapi/dns_dnshome.sh [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+#
+# export DNSHOME_Subdomain=""
+# export DNSHOME_SubdomainPassword=""
+
+# Usage: add subdomain.ddnsdomain.tld "XKrxpRBosdIKFzxW_CT3KLZNf6q0HG9i01zxXp5CPBs"
+# Used to add txt record
+dns_dnshome_add() {
+  txtvalue=$2
+
+  DNSHOME_Subdomain="${DNSHOME_Subdomain:-$(_readaccountconf_mutable DNSHOME_Subdomain)}"
```
Comment (Neilpang) [inline dnsapi/dns_dnshome.sh:19]: please use `_readdomainconf` instead:

```
DNSHOME_Subdomain="${DNSHOME_Subdomain:-$(_readdomainconf DNSHOME_Subdomain)}"

DNSHOME_SubdomainPassword="${DNSHOME_SubdomainPassword:-$(_readdomainconf DNSHOME_SubdomainPassword)}"

```

File: dnsapi/dns_dnshome.sh [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+  if [ -z "$DNSHOME_Subdomain" ] || [ -z "$DNSHOME_SubdomainPassword" ]; then
+    DNSHOME_Subdomain=""
+    DNSHOME_SubdomainPassword=""
+    _err "Please specify/export your dnsHome.de Subdomain and Password"
+    return 1
+  fi
+
+  #save the credentials to the account conf file.
+  _saveaccountconf_mutable DNSHOME_Subdomain "$DNSHOME_Subdomain"
```
Comment (Neilpang) [inline dnsapi/dns_dnshome.sh:30]: please use `_savedomainconf` instead too.

**Pull request merged**